form/multiple_buttons.rst

```diff
@@ -35,3 +35,9 @@ Or you can get the button's name by using the
     if ($form->getClickedButton() && 'saveAndAdd' === $form->getClickedButton()->getName()) {
         // ...
     }
+    
+If you have multiple buttons with the same name, each nested in multiple subforms, use direct reference instead the name to get the right button
```

Comment:
I would clarify here that this is related to a form name used multiple times. And we could also just add it inside the code block above. Maybe something like this:

```rst
Or you can get the button's name by using the
:method:`Symfony\\Component\\Form\\Form::getClickedButton` method of the form::

    if ($form->getClickedButton() && 'saveAndAdd' === $form->getClickedButton()->getName()) {
        // ...
    }

    // if you have multiple buttons with the same name, compare the object references instead
    if ($form->getClickedButton() === $form->get('saveAndAdd')){
        // ...
    }
```

Reply:
Yes but multiple buttons with the same name in the same form is not possible, so I wanted to clarify that it was in nested form situation.

Reply:
Maybe using @xabbuh's proposal + a small note about the "same name"?